modules/misc/xdg-user-dirs.nix

```
@@ -33,57 +33,71 @@ in {
     # https://gitlab.freedesktop.org/xdg/xdg-user-dirs/blob/master/man/user-dirs.dirs.xml
 
     desktop = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Desktop";
       description = "The Desktop directory.";
     };
 
     documents = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Documents";
       description = "The Documents directory.";
     };
 
     download = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Downloads";
       description = "The Downloads directory.";
     };
 
     music = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Music";
       description = "The Music directory.";
     };
 
     pictures = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Pictures";
       description = "The Pictures directory.";
     };
 
     publicShare = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Public";
       description = "The Public share directory.";
     };
 
     templates = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Templates";
       description = "The Templates directory.";
     };
 
     videos = mkOption {
-      type = types.str;
+      type = with types; either str path;
+      apply = toString;
       default = "$HOME/Videos";
       description = "The Videos directory.";
     };
 
     extraConfig = mkOption {
-      type = with types; attrsOf str;
+      type = with types; attrsOf (either str path);
+      apply = builtins.mapAttrs (_: toString);
```

```suggestion
      type = with types; attrsOf (coercedTo path toString str);
```
minor nitpick, I think `coercedTo` is better suited here as only type coercion is happening, rather than transformation on the option value.

---

Thank you. I wasn't aware of the `coercedTo` type.